src/compas_timber/_fabrication/lap.py

```diff
@@ -0,0 +1,511 @@
+            raise ValueError("Width must be between 0.0 and 50000.")
+        self._width = width
+
+    @property
+    def depth(self):
+        return self._depth
+
+    @depth.setter
+    def depth(self, depth):
+        if depth > 50000.0 or depth < -50000.0:
+            raise ValueError("Depth must be between -50000 and 50000.")
+        self._depth = depth
+
+    @property
+    def lead_angle_parallel(self):
+        return self._lead_angle_parallel
+
+    @lead_angle_parallel.setter
+    def lead_angle_parallel(self, lead_angle_parallel):
+        if not isinstance(lead_angle_parallel, bool):
+            raise ValueError("Lead angle parallel must be a boolean.")
+        self._lead_angle_parallel = lead_angle_parallel
+
+    @property
+    def lead_angle(self):
+        return self._lead_angle
+
+    @lead_angle.setter
+    def lead_angle(self, lead_angle):
+        if lead_angle > 179.9 or lead_angle < 0.1:
+            raise ValueError("Lead angle must be between 0.1 and 179.9.")
+        self._lead_angle = lead_angle
+
+    @property
+    def lead_inclination_parallel(self):
+        return self._lead_inclination_parallel
+
+    @lead_inclination_parallel.setter
+    def lead_inclination_parallel(self, lead_inclination_parallel):
+        if not isinstance(lead_inclination_parallel, bool):
+            raise ValueError("Lead inclination parallel must be a boolean.")
+        self._lead_inclination_parallel = lead_inclination_parallel
+
+    @property
+    def lead_inclination(self):
+        return self._lead_inclination
+
+    @lead_inclination.setter
+    def lead_inclination(self, lead_inclination):
+        if lead_inclination > 179.9 or lead_inclination < 0.1:
+            raise ValueError("Lead inclination must be between 0.1 and 179.9.")
+        self._lead_inclination = lead_inclination
+
+    @property
+    def machining_limits(self):
+        return self._machining_limits
+
+    @machining_limits.setter
+    def machining_limits(self, machining_limits):
+        if not isinstance(machining_limits, dict):
+            raise ValueError("Machining limits must be a dictionary.")
+        for key, value in machining_limits.items():
+            if key not in MachiningLimits.EXPECTED_KEYS:
+                raise ValueError("The key must be one of the following: ", {self.EXPECTED_KEYS})
+            if not isinstance(value, bool):
+                raise ValueError("The values must be a boolean.")
+        self._machining_limits = machining_limits
+
+
+    ########################################################################
+    # Alternative constructors
+    ########################################################################
+
+    @classmethod
+    def from_two_planes_and_beam(cls, planes, beam, depth, ref_side_index=0):
+        """Create a Lap instance from two planes and a beam. The planes should be parallel to each other and their normals should be facing in the opposite direction.
```

Comment:
a couple of questions here
these two planes are the side walls of the lap, right?
since there's quite a strict requirement, would it maybe be more robust if caller provides just one of the planes and the desired distance between them? 
the we could create the second plane ourselves, that's quite trivial I guess.